Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskRotateActor.cpp

```diff
@@ -30,18 +27,67 @@ EStateTreeRunStatus FAIExtStateTreeTaskRotateActor::Tick( FStateTreeExecutionCon
 {
     auto & instance_data = context.GetInstanceData< FInstanceDataType >( *this );
 
-    const auto current_rotation = instance_data.ActorToRotate->GetActorRotation();
+    const auto current_rotation = instance_data.Actor->GetActorRotation();
 
-    if ( current_rotation.Equals( instance_data.TargetRotation, 0.01f ) )
+    if ( instance_data.bContinuousRotation )
+    {
+        UpdateTargetRotation( context );
+    }
+
+    auto rotation = FMath::RInterpConstantTo( current_rotation, instance_data.TargetRotation, delta_time, instance_data.RotationSpeed );
+    auto rotation_matches = true;
+
+    if ( !instance_data.bUpdatePitch )
+    {
+        rotation.Pitch = current_rotation.Pitch;
+    }
+    else
     {
-        if ( instance_data.bFinishTaskWhenRotationIsComplete )
-        {
-            return EStateTreeRunStatus::Succeeded;
-        }
+        rotation_matches &= FMath::IsNearlyEqual( rotation.Pitch, instance_data.TargetRotation.Pitch, 0.01f );
```

Comment:
Maybe it could be useful if you could specify the treshold in the STT?

Reply:
it will over-complexify the task in the editor. we just want to know if its roughly equal